bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/flamingos2/Flamingos2.java

```diff
@@ -852,8 +856,25 @@ public static double getFpuChangeOverheadSec() {
         return CommonStepCalculator.instance.calc(cur, prev).addAll(times);
     }
 
+    private static final double IMAGING_SETUP_TIME_OIWFS = 60 * 15;
+
+    private static final double IMAGING_SETUP_TIME_PWFS2 = 60 * 6;
+
+    private static final ItemKey GUIDE_WITH_OIWFS_KEY = new ItemKey(TELESCOPE_KEY, TargetObsCompConstants.GUIDE_WITH_OIWFS_PROP);
+
     public static double getImagingSetupSec(ISPObservation obs) {
-        return 60 * (usesF2Oiwfs(obs) ? 15 : 6);
+        return usesF2Oiwfs(obs) ? IMAGING_SETUP_TIME_OIWFS : IMAGING_SETUP_TIME_PWFS2;
+    }
+
+
+    public static double getImagingSetupSec(Config[] conf) {
+        if (conf[0].containsItem(GUIDE_WITH_OIWFS_KEY)) {
```

Comment:
Since these methods work with just one `Config` instance, we should probably change the signature to reflect that fact.  This is, push up the extraction of the first step as much as possible.

Reply:
Thanks, I have fixed this for all methods using conf[0].